tardis/adapters/sites/satellite.py

```diff
@@ -0,0 +1,381 @@
+            self.cached_status_coroutines[remote_resource_uuid] = AsyncCacheMap(
+                partial(self._get_status, remote_resource_uuid),
+                max_age=self.max_age,
+            )
+        await self.cached_status_coroutines[remote_resource_uuid].update_status(
+            force=force
+        )
+        return self.cached_status_coroutines[remote_resource_uuid]
+
+    async def _get_status(self, remote_resource_uuid: str) -> dict:
+        """
+        Return host data together with custom parameters and power details.
+
+        :param remote_resource_uuid: Satellite identifier of the host.
+        :type remote_resource_uuid: str
+        :return: Satellite host data enriched with parameters and power state.
+        :rtype: dict
+        """
+        async with aiohttp.ClientSession(auth=self.auth) as session:
+            host_url = self._host_url(remote_resource_uuid)
+            main_task = self._request(session, "GET", host_url)
+            params_task = self._request(session, "GET", f"{host_url}/parameters")
+            power_task = self._request(session, "GET", f"{host_url}/power")
+            main_response, param_response, power_response = await asyncio.gather(
+                main_task, params_task, power_task
+            )
+
+        # Flatten custom parameters for simpler lookups in later calls.
+        parameters = {}
+        for param in param_response.get("results", []):
+            name = param.get("name")
+            if not name:
+                continue
+            if "value" in param:
+                parameters[name] = param["value"]
+            if "id" in param:
+                parameters[f"{name}_id"] = param["id"]
+
+        main_response["parameters"] = parameters
+        main_response["power"] = power_response
+        return main_response
+
+    async def set_power(self, state: str, remote_resource_uuid: str) -> dict:
+        """
+        Set the power state of a host and update its cached status.
+
+        :param state: Desired power state as understood by the Satellite API ["on"|"off"].
+        :type state: str
+        :param remote_resource_uuid: Satellite identifier of the host.
+        :type remote_resource_uuid: str
+        :return: Raw response from the Satellite power endpoint.
+        :rtype: dict
+        """
+
+        if state not in ("on", "off"):
+            raise ValueError(f"Invalid power state {state}")
+
+        async with aiohttp.ClientSession(auth=self.auth) as session:
+            logger.info(f"Set power {state} for {remote_resource_uuid}")
+            power_action_result = await self._request(
+                session,
+                "PUT",
+                f"{self._host_url(remote_resource_uuid)}/power",
+                json={"power_action": state},
+            )
+            await self.get_status(remote_resource_uuid, force=True)
+        return power_action_result
+
+    async def get_next_uuid(self) -> str:
```

Comment:
Maybe a different name of the function would be more descriptive. `get_next_host` or `get_available_host`, or similar.